pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -113,6 +113,8 @@ def attach_dataloaders(
     def attach_datamodule(
         self, model: 'pl.LightningModule', datamodule: Optional['pl.LightningDataModule'] = None
     ) -> None:
+        datamodule = datamodule or getattr(model, 'datamodule', None)
+
```

Comment:
@carmocca removed this in a recent PR: 
https://github.com/PyTorchLightning/pytorch-lightning/pull/8002/files#diff-906ad7bb3bf022c44bada799c11c2b45ef0c12c3de2cc743d3078bd95cd0d778

Reply:
@awaelchli there's a discussion going on in the linked issue